src/Input/Silk.NET.Input.Common/GamepadExtensions.cs

```diff
@@ -22,80 +22,112 @@ public static class GamepadExtensions
 
             throw new PlatformNotSupportedException("This button is not supported by the input backend.");
         }
-
+        
+        private static Thumbstick FirstThumbstick(IGamepad gamepad, int index)
+        {
+            if (gamepad.Thumbsticks.Count >= index)
+            {
+                throw new PlatformNotSupportedException("This thumbstick is not supported by the input backend.");
+            }
+            
+            return gamepad.Thumbsticks[index];
+        }
```

Comment:
This really isn't necessary, the indexer throwing an ArgumentOutOfRangeException is fit for purpose - we don't need our own bounds check.

Reply:
I wanted to replicate the same error since this does end up getting exposed to the user. "This thumbstick is not supported" is a lot more clear than "There was an out of range error somewhere".

Reply:
This is what documentation is for.

Reply:
(Plus it’s usually best to have strongly-typed, predictable errors :))